### DISCUSSION TITLE
pod install is failing when running renovate cli

### DISCUSSION THREAD

**Discussion by asoneji** [renovatebot/renovate#13174] [Request Help] [answered]
pod install is failing when running renovate cli

**`renovate.json`:**
```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "exposeAllEnv": true,
  "recreateClosed": true
}

```
**Renovate version:** `31.5.0`
**Running renovate cli:**
`LOG_LEVEL=debug yarn start MY_REPO token=MY_TOKEN --dry-run=true --endpoint="https://github.GITHUB_ENTERPRISE_URL.com/api/v3/"`

**Error:**
##### File name: sample-ios-project/Podfile.lock

```
/Users/asoneji/.rvm/rubies/ruby-2.7.0/lib/ruby/2.7.0/rubygems.rb:275:in `find_spec_for_exe': can't find gem cocoapods (>= 0.a) with executable pod (Gem::GemNotFoundException)
	from /Users/asoneji/.rvm/rubies/ruby-2.7.0/lib/ruby/2.7.0/rubygems.rb:294:in `activate_bin_path'
	from /Users/asoneji/.rvm/gems/ruby-2.7.0/bin/pod:23:in `<main>'
	from /Users/asoneji/.rvm/gems/ruby-2.7.0/bin/ruby_executable_hooks:22:in `eval'
	from /Users/asoneji/.rvm/gems/ruby-2.7.0/bin/ruby_executable_hooks:22:in `<main>'

```

**Note:**
* I also have created an empty `.npmrc` file
* pod install locally works

**Comment by viceice** [comment]
You need to manually install cocoa pods and put onto path.

**Reply by asoneji** [reply]
I do have cocoa pods installed. Do I need to do anything else? Asking as `pod install` works as expected when I run it from the command line.

```
$ pod --version
1.11.2
$ which pod
/Users/asoneji/.rvm/gems/ruby-2.7.0/bin/pod
$
```

**Reply by viceice** [reply]
that path isn't on default `PATH`, renovate isn't executing a shell when calling pod, so there are probably some missing env values from `rvm`

**Reply by viceice** [reply]
you should try to use docker binary source

**Reply by asoneji** [reply]
I have added `"exposeAllEnv": true,` to my `renovate.json` which I thought will pass all my local patht to renovate so should work as it works locally.

I have `export PATH="$PATH:$HOME/.rvm/bin"` in my `.zshrc`.

I can't use the docker image as we have have depdency they require Xcode and so need to run locally on my machine and not in the linux docker image.

**Reply by asoneji** [reply]
when I clone `git@github.com:renovatebot/renovate.git` and add `config.js` it works.

**`config.js`:**
```
module.exports = {
  exposeAllEnv: true
};
```

Command: ` LOG_LEVEL=debug yarn start MY_REPO token=MY_TOKEN --dry-run=true --endpoint="https://github.GITHUB_ENTERPRISE_URL.com/api/v3/"`

With this, it is able to do `pod install` without any issue and create the PR without any errors.


How can I pass `exposeAllEnv: true` similar to above when running renovate cli?

**Reply by viceice** [reply]
exposeAllEnv can only be used in global bot config, you can pass it via emv.

You can't use any global config in renovate.json. they can only be used via enc, cli or config.js.

**Reply by asoneji** [reply]
If I have installed renovate globally like `npm renovate -g install` and using the renovate command `LOG_LEVEL=debug yarn start MY_REPO token=MY_TOKEN --dry-run=true --endpoint="https://github.GITHUB_ENTERPRISE_URL.com/api/v3/"` how do I pass/set `exposeAllEnvtrue` to `true`?

**Reply by rarkins** [reply]
`--expose-all-env=true` or add a `config.js` file to your `pwd` as described in the docs

**Reply by asoneji** [reply]
got it thanks! I did try adding `--exposeAllEnvtrue=true` and it said it was invalid options and didn't see that it is called `--expose-all-env`.

I will try this and or setting `config.js` in PWD. I was also was able to do this using `RENOVATE_CONFIG_FILE="/tmp/renovate-config.js"`

**Comment by asoneji** [comment, marked as answer]
pass `--expose-all-env=true` or add a `config.js` file to your pwd as described in the docs
